src/App.tsx

```diff
@@ -1,27 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum GoodsFilters {
+  ALL = 'all',
+  FIVE = 'five',
+  RED = 'red',
+  NOTHING = '',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+function getVisibleGoods(good: GoodsFilters): Promise<Good[]> | null {
+  switch (good) {
+    case GoodsFilters.ALL:
+      return getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    case GoodsFilters.FIVE:
+      return get5First();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    case GoodsFilters.RED:
+      return getRedGoods();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    default:
+      return null;
+  }
+}
 
-    <GoodsList goods={[]} />
-  </div>
-);
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [visibleGoods, setVisibleGoods] = useState(GoodsFilters.NOTHING);
+
+  useEffect(() => {
+    getVisibleGoods(visibleGoods)?.then(setGoods);
+  }, [visibleGoods]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          setVisibleGoods(GoodsFilters.ALL);
+        }}
```

Comment:
You can create memoize value to prevent make request each time when you click on button, for example:
`
const allTodos = useMemo(() => getAll(), [])
const firstFiveTodos= useMemo(() => get5First(), [])
`